### ISSUE TITLE
Running sql-server from an empty state make inconsistent repository

### ISSUE THREAD

**Issue by guilhem** [dolthub/dolt#7873]
Running sql-server on an empty dir make dolt in an inconsistent state.

sql-server is working well, but `dolt ls` make error:
```
The current directories repository state is invalid
open /var/lib/dolt/.dolt/repo_state.json: no such file or directory
```

It's also impossible to manipulate remotes, etc.

It's also imposible to dolt init:
```
Failed to initialize directory as a data repo. .dolt directory already exists at '.'
```

**Comment by timsehn** [comment]
Thanks for the report. I'll try and reproduce. I assume this is on a *nix platfiorm?

**Comment by fulghum** [comment]
Hi @guilhem, thanks for taking the time to report this one. I did an audit of how Dolt CLI commands behave when you run them in a directory that is being used for a dolt sql-server, but has not been initialized as a Dolt database yet, and I left the results in https://github.com/dolthub/dolt/issues/7875. 

Some commands display a better error message (i.e. "The current directory is not a valid dolt repository."), but many other commands display a confusing error message like the one you reported (i.e. "The current directories repository state is invalid"). I'm working on some changes to make the CLI commands consistently report "The current directory is not a valid dolt repository". 

For `dolt init`, I'm going to customize the error message with some better instructions. Since a running `dolt sql-server` won't pick up changes properly if you run `dolt init` on the directory it's running from, we'll output an error message telling customers that a `dolt sql-server` is running from the same directory, and that it needs to be stopped before that directory can be initialized. 

Let us know if you think more is needed to make those error messages more helpful and easier to understand.

**Comment by timsehn** [comment]
This should be fixed by #7876 and the fix to dolt remotes #7622 when @macneale4 has it. I'll resolve as we decomposed it into multiple bugs.

**Comment by LaPingvino** [comment]
Comment for future me because I ran into this for a local repository that I accidentally found back, you can create a remote to a file: destination, dolt push --all to it and clone from there to a new location to cleanly rebuild the repository without issues.
